linked_list/linked_list.py

```diff
@@ -0,0 +1,246 @@
+from __future__ import print_function  # For Python 2
+
+
+class Node(object):
+    def __init__(self, data=None, next_=None):
+        # Initialises Node
+        self.data = data
+        self.next = next_
+
+    def __str__(self):
+        return str(self.data)
+
+
+class Linked_list(object):
+    def __init__(self):
+        # Initialises Linked List.
+        self.head = None
+        self.size = 0
+
+    def __str__(self):
+        n = self.head
+        print('Linked List : [', end="")
+        while n:
+            if n.next:
+                print(n, end=", ")
+            else:
+                print(n, end="")
+            n = n.next
+        print(']')
+
+    def __len__(self):
+        print('Size :', self.size)
+
+    def add_front(self, data):
+        # Inserts element at first position.
+        self.add(data, 0)
+
+    def add_last(self, data):
+        # Inserts element at last position.
+        self.add(data, self.size)
+
+    def add(self, data, index):
```

Comment:
Please try to understand this can't be done as self is not defined till function starts.